pymchelper/readers/shieldhit/general.py

```diff
@@ -122,6 +124,18 @@ def get_reader(self):
 
         # TODO add ZIP file unpacking
 
+        # Trying to decompress bdz-file. If it works, saves a temporary bdo-file which will be used for reading later
+        # zip_name is the addition to the file name for the new bdo-file, compared to the original: nothing in case
+        # original is already bdo, and "_TEMP.bdo" in case original was a bdz.
+        with open(self.filename) as f:
```

Comment:
You should tell python that it is a *binary* stream opened for reading. Try `open(filename, "rb")`.